homeassistant/components/zha/core/channels/__init__.py

```diff
@@ -5,404 +5,314 @@
+
+    async def async_configure(self) -> None:
+        """Configure claimed channels."""
+        await self._execute_channel_tasks("async_configure")
+
+    async def _execute_channel_tasks(self, func_name: str, *args: Any) -> None:
+        """Add a throttled channel task and swallow exceptions."""
+
+        async def _throttle(coro):
+            async with self._channels.semaphore:
+                return await coro
+
+        channels = [*self.claimed_channels.values(), *self.relay_channels.values()]
+        tasks = [_throttle(getattr(ch, func_name)(*args)) for ch in channels]
+        results = await asyncio.gather(*tasks, return_exceptions=True)
+        for channel, outcome in zip(channels, results):
+            if isinstance(outcome, Exception):
+                channel.warning("'%s' stage failed: %s", func_name, str(outcome))
+                continue
+            channel.debug("'%s' stage succeeded", func_name)
 
-    def log(self, level, msg, *args):
-        """Log a message."""
-        msg = f"[%s:ZDO](%s): {msg}"
-        args = (self._zha_device.nwk, self._zha_device.model) + args
-        _LOGGER.log(level, msg, *args)
+    @callback
+    def async_new_entity(
+        self,
+        component: str,
+        entity: zha_typing.CALLABLE_T,
+        unique_id: str,
+        channels: List[zha_typing.ChannelType],
+    ):
+        """Signal new entity addition."""
+        self._channels.async_new_entity(component, entity, unique_id, channels)
 
+    @callback
+    def async_send_signal(self, signal: str, *args: Any) -> None:
+        """Send a signal through hass dispatcher."""
+        self._channels.async_send_signal(signal, *args)
 
-class EventRelayChannel(ZigbeeChannel):
-    """Event relay that can be attached to zigbee clusters."""
+    @callback
+    def claim_channels(self, channels: List[zha_typing.ChannelType]) -> None:
+        """Claim a channel."""
+        self.claimed_channels.update({ch.id: ch for ch in channels})
 
-    CHANNEL_NAME = CHANNEL_EVENT_RELAY
+    @callback
+    def unclaimed_channels(self) -> List[zha_typing.ChannelType]:
+        """Return a list of available (unclaimed) channels."""
+        claimed = set(self.claimed_channels)
+        available = set(self.all_channels)
+        return [self.all_channels[chan_id] for chan_id in (available - claimed)]
 
     @callback
-    def attribute_updated(self, attrid, value):
-        """Handle an attribute updated on this cluster."""
-        self.zha_send_event(
-            self._cluster,
-            SIGNAL_ATTR_UPDATED,
+    def zha_send_event(self, event_data: Dict[str, Union[str, int]]) -> None:
+        """Relay events to hass."""
+        self._channels.zha_send_event(
             {
-                "attribute_id": attrid,
-                "attribute_name": self._cluster.attributes.get(attrid, ["Unknown"])[0],
-                "value": value,
-            },
+                const.ATTR_UNIQUE_ID: self.unique_id,
```

Comment:
Hrm one of my comments vanished... when we delegate to `_channels` here we are also including `const.ATTR_UNIQUE_ID: self.unique_id` in the event again. Is the value the same in both places?

Reply:
Not the same, but any attribute present in the `data`  in the top of the chain, takes precedence over the same attribute added later. In other words `const.ATTR_UNIQUE_ID: self.unique_id` is attribute added only if we don't already have one. 
So channel event has it's `unique_id` there and I still have an `unique_id` if I originate an  event at `ChannelPool` or `Channels` level.

Reply:
OK, that makes sense